src/script/interpreter.h

```diff
@@ -133,12 +153,14 @@ class TransactionSignatureChecker : public BaseSignatureChecker
     const CTransaction* txTo;
     unsigned int nIn;
     const CAmount amount;
+    mutable CachedHashes* cachedHashes;
```

Comment:
> it is CachedHashes is modified by SignatureHash

I don't think so? cacheHashes is not modified. *cacheHashes is.

If you have a const TransactionSignatureChecker method, its cacheHashes
variable is of type const pointer to non-const CachedHashes. It means the
pointer can't be changed, but the object pointed to can be.